src/config/validation.rs

```diff
@@ -189,6 +192,137 @@ pub fn check_outputs(config: &ConfigBuilder) -> Result<(), Vec<String>> {
+
+    if configured_disk_buffers.is_empty() {
+        return Ok(());
+    }
+
+    // Now query all the mountpoints on the system, and get their total capacity. We also have to
+    // sort the mountpoints from longest to shortest so we can find the longest prefix match for
+    // each buffer data directory by simply iterating from beginning to end.
+    let mountpoints = heim::disk::partitions()
+        .and_then(|stream| stream.try_collect::<Vec<_>>().and_then(process_partitions))
+        .or_else(|_| {
+            heim::disk::partitions_physical()
+                .and_then(|stream| stream.try_collect::<Vec<_>>().and_then(process_partitions))
+        })
+        .await;
+
+    let mountpoints = match mountpoints {
+        Ok(mut mountpoints) => {
+            mountpoints.sort_by(|m1, _, m2, _| m2.cmp(m1));
+            mountpoints
+        }
+        Err(e) => {
+            warn!(
+                cause = %e,
+                message = "Failed to query disk partitions. Cannot ensure that buffer size limits are within physical storage capacity limits.",
+            );
+            return Ok(());
+        }
+    };
+
+    // Now build a mapping of buffer IDs/usage configuration to the mountpoint they reside on.
+    let mountpoint_buffer_mapping =
+        configured_disk_buffers
+            .into_iter()
+            .fold(HashMap::new(), |mut mappings, usage| {
+                let canonicalized_data_dir = usage
+                    .data_dir()
+                    .canonicalize()
+                    .unwrap_or_else(|_| usage.data_dir().to_path_buf());
+                let mountpoint = mountpoints
+                    .keys()
+                    .find(|mountpoint| canonicalized_data_dir.starts_with(mountpoint));
+
+                match mountpoint {
+                    None => warn!(
+                        buffer_id = usage.id().id(),
+                        data_dir = usage.data_dir().to_string_lossy().as_ref(),
+                        canonicalized_data_dir = canonicalized_data_dir.to_string_lossy().as_ref(),
+                        message = "Found no matching mountpoint for buffer data directory.",
+                    ),
+                    Some(mountpoint) => {
+                        mappings
+                            .entry(mountpoint.clone())
+                            .or_insert_with(Vec::new)
+                            .push(usage);
+                    }
+                }
+
+                mappings
+            });
+
+    // Finally, we have a mapping of disk buffers, based on their underlying mountpoint. Go through
+    // and check to make sure the sum total of `max_size` for all buffers associated with each
+    // mountpoint does not exceed that mountpoint's total capacity.
+    //
+    // We specifically do not do any sort of warning on free space because that has to be the
+    // responsibility of the operator to ensure there's enough total space for all buffers present.
```

Comment:
This sticks out/reads funny to me, but I can't exactly put my finger on it.

Reply:
Like the wording itself or the implication of the statement?

Reply:
The wording - it's definitely fine to merge as is, I can't think of how or even why I'd exactly change it. If you happen to have a thought immediately about it happy to check it out again but otherwise 🚀